File: BackEnd_Learn/Java/auth/src/main/java/io/github/leeseojune53/auth/controller/UserController.java
package io.github.leeseojune53.auth.controller;

import io.github.leeseojune53.auth.dto.UserAuthRequest;
import io.github.leeseojune53.auth.dto.UserRegisterRequest;
import io.github.leeseojune53.auth.service.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.*;

@RestController
@RequiredArgsConstructor
public class UserController {

    private final UserService userService;

    @PostMapping("/register")
    @ResponseStatus(HttpStatus.CREATED)
    public void register(@RequestBody UserRegisterRequest registerRequest) {
        userService.userRegister(registerRequest);
    }

    @PostMapping("/auth")
    public String token(@RequestBody UserAuthRequest authRequest) {
        return userService.getAccessToken(authRequest);
    }

    @GetMapping("/user/info")
    public String user() {
        return "user role";
    }

}

File: BackEnd_Learn/kotlin/auth/src/main/kotlin/io/github/leeseojune53/auth/controller/UserController.kt
package io.github.leeseojune53.auth.controller

import io.github.leeseojune53.auth.dto.UserRegisterRequest
import io.github.leeseojune53.auth.service.UserService
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestController


@RestController
class UserController(
    private val userService: UserService
) {

    @PostMapping("/register")
    @ResponseStatus(HttpStatus.CREATED)
    fun register(@RequestBody userRegisterRequest: UserRegisterRequest) {
        userService.register(userRegisterRequest)
    }

}